native/src/main/java/io/ballerina/stdlib/sql/nativeimpl/ExecuteProcessor.java

```diff
@@ -208,28 +209,18 @@ private static Object nativeBatchExecuteExecutable(BObject client, BArray paramS
                 } else {
                     statement = connection.prepareStatement(sqlQuery, Statement.NO_GENERATED_KEYS);
                 }
-
-                for (Object[] param : parameters) {
-                    statementParameterProcessor.setParams(connection, statement, param);
-                    statement.addBatch();
-                }
-
-                int[] counts = statement.executeBatch();
-
                 if (sqlDatasource.getBatchExecuteGKFlag() && !isDdlStatement(sqlQuery)) {
-                    resultSet = statement.getGeneratedKeys();
+                    isGeneratedKeys = true;
                 }
-                for (int count : counts) {
-                    Map<String, Object> resultField = new HashMap<>();
-                    resultField.put(Constants.AFFECTED_ROW_COUNT_FIELD, count);
-                    Object lastInsertedId = null;
-                    if (resultSet != null && resultSet.next()) {
-                        lastInsertedId = getGeneratedKeys(resultSet);
+                for (int i = 0; i < parameters.size(); i++) {
```

Comment:
```suggestion
                for (int paramIndex = 0; paramIndex < parameters.size(); paramIndex++) {
```

Reply:
Still have't changed the variable name

Reply:
Updated in [784c70d](https://github.com/ballerina-platform/module-ballerina-sql/pull/629/commits/784c70d07880677e0082654dc2caa3fe107cfe7a)